app/src/main/java/org/gdg/frisbee/android/event/EventActivity.java

```diff
@@ -53,16 +70,86 @@ public void onCreate(Bundle savedInstanceState) {
         mViewPager.setAdapter(mViewPagerAdapter);
         mTabLayout.setupWithViewPager(mViewPager);
 
-        mEventId = getIntent().getStringExtra(Const.EXTRA_EVENT_ID);
+        mEventId = getEventIdFrom(getIntent());
         String section = getIntent().getStringExtra(Const.EXTRA_SECTION);
         if (EventPagerAdapter.SECTION_OVERVIEW.equals(section)) {
             mViewPager.setCurrentItem(0);
         }
     }
 
+    private String getEventIdFrom(Intent intent) {
+        return intent.getStringExtra(Const.EXTRA_EVENT_ID);
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        recordStartPageView();
+    }
+
+    @Override
+    protected void onStop() {
+        recordEndPageView();
+        super.onStop();
+    }
+
     protected String getTrackedViewName() {
         return "Event/" + getResources().getStringArray(R.array.event_tabs)[getCurrentPage()] 
-                + "/" + getIntent().getStringExtra(Const.EXTRA_EVENT_ID);
+                + "/" + getEventIdFrom(getIntent());
+    }
+
+    private void recordStartPageView() {
+        App.getInstance().getGdgXHub().getEventDetail(
+                mEventId, new Callback<EventFullDetails>() {
+                    @Override
+                    public void success(EventFullDetails eventFullDetails, retrofit.client.Response response) {
+                        mEventFullDetails = eventFullDetails;
+                        Action viewAction = createAppIndexAction(eventFullDetails.getTitle(), mEventId);
+                        PendingResult<Status> result = AppIndex.AppIndexApi.start(getGoogleApiClient(), viewAction);
+                        result.setResultCallback(appIndexApiCallback("start " + viewAction));
+                    }
+
+                    @Override
+                    public void failure(RetrofitError error) {
+
+                    }
+                }
+        );
+    }
+
+    private void recordEndPageView() {
+        if (mEventFullDetails != null) {
+            Action viewAction = createAppIndexAction(mEventFullDetails.getTitle(), mEventId);
+            PendingResult<Status> result = AppIndex.AppIndexApi.end(getGoogleApiClient(), viewAction);
+            result.setResultCallback(appIndexApiCallback("end " + viewAction));
+        }
+    }
+
+    private ResultCallback<Status> appIndexApiCallback(final String label) {
+        return new ResultCallback<Status>() {
+            @Override
+            public void onResult(Status status) {
+                if (status.isSuccess()) {
+                    Timber.d(
```

Comment:
It looks like we have duplicate code here. I am thinking to extract these App Indexing related classes to a super class.